package/src/core/define-integration.ts

```diff
@@ -43,12 +43,15 @@ export const defineIntegration = <
 		name: string;
 		options: z.output<TOptionsSchema>;
 	}) => ExtendedHooks<TPlugins>;
-}): ((options?: z.input<TOptionsSchema>) => AstroIntegration) => {
-	return (_options: z.input<TOptionsSchema> = {}) => {
-		const parsedOptions = (optionsSchema || z.record(z.any())).safeParse(
-			_options,
-			{ errorMap },
-		);
+}): ((
+	...args: undefined extends z.input<TOptionsSchema>
+		? [options?: z.input<TOptionsSchema>]
+		: [options: z.input<TOptionsSchema>]
```

Comment:
Beautiful

Reply:
It's all thanks to you! Learned the trick while doing the i18n integration ;)